src/Symfony/Component/HttpKernel/DataCollector/RequestDataCollector.php

```diff
@@ -40,12 +38,8 @@ public function __construct()
     public function collect(Request $request, Response $response, \Exception $exception = null)
     {
         $responseHeaders = $response->headers->all();
-        $cookies = array();
         foreach ($response->headers->getCookies() as $cookie) {
-            $cookies[] = $this->getCookieHeader($cookie->getName(), $cookie->getValue(), $cookie->getExpiresTime(), $cookie->getPath(), $cookie->getDomain(), $cookie->isSecure(), $cookie->isHttpOnly());
-        }
-        if (count($cookies) > 0) {
-            $responseHeaders['Set-Cookie'] = $cookies;
+            $responseHeaders['set-cookie'][] = (string) $cookie;
         }
```

Comment:
Above should apply to 3.1 & 3.2 as well.

In master we can remove this block due #20569. I covered this with #20567 and should be a separate PR otherwise (basically forgotten in #20569).